File: metalanguage/rosetta/message_handovers/agent.py
class Agent:
    def __init__(self, name):
        self.name = name
        self.counter = 0

def send_message(sender, receiver):
    receiver.counter += 1
    print(f"{sender.name} -> {receiver.name}: {receiver.counter}")

if __name__ == "__main__":
    alice = Agent("Alice")
    bob = Agent("Bob")
    send_message(alice, bob)
    send_message(alice, bob)

File: metalanguage/rosetta/message_handovers/agent.rs
struct Agent {
    name: String,
    counter: u32,
}

fn send_message(sender: &Agent, receiver: &mut Agent) {
    receiver.counter += 1;
    println!("{} -> {}: {}", sender.name, receiver.name, receiver.counter);
}

fn main() {
    let alice = Agent { name: "Alice".to_string(), counter: 0 };
    let mut bob = Agent { name: "Bob".to_string(), counter: 0 };
    send_message(&alice, &mut bob);
    send_message(&alice, &mut bob);
}
